### PR TITLE
21-0845 - Add automated front-end tests

### PR DESCRIPTION
**Pull request by tlei123**
## Summary

- Add automated front-end tests to PI Release Auth (21-0845).
- Team: VA Product Forms
- Flipper not implemented yet

## Related issue(s)

- _Link to ticket created in va.gov-team-forms repo_
department-of-veterans-affairs/va.gov-team-forms#367

## Testing done

- All unit-tests and e2e-spec passed locally.
- E2E-spec disabled in CI until CMS landing-page is published.

### CODE REVIEW

File: src/applications/simple-forms/21-0845/config/form.js [hunk 1]
```diff
@@ -81,8 +81,8 @@ const formConfig = {
           title: 'Who’s submitting this authorization?',
           // we want req'd fields prefilled for LOCAL testing/previewing
           // one single initialData prop here will suffice for entire form
-          initialData:
-            !!mockData && environment.isLocalhost() ? mockData : undefined,
+          // initialData:
+          //   !!mockData && environment.isLocalhost() ? mockData : undefined,
```
Comment (gmrabian) [inline src/applications/simple-forms/21-0845/config/form.js:85]: did you mean to leave all of this commented out?

File: src/applications/simple-forms/21-0845/tests/pages/releaseDuration.unit.spec.jsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  testNumberOfFields,
+} from '../../../shared/tests/pages/pageTests.spec';
+import formConfig from '../../config/form';
+
+const {
+  schema,
+  uiSchema,
+} = formConfig.chapters.infoReleaseChapter.pages.releaseDurationPage;
+
+const pageTitle = 'Who’s submitting this authorization?';
```
Comment (gmrabian) [inline src/applications/simple-forms/21-0845/tests/pages/releaseDuration.unit.spec.jsx:12]: is this the right title?

**Review by gmrabian — COMMENTED** [review]
a few comments but looks good!

File: src/applications/simple-forms/21-0845/tests/pages/releaseDuration.unit.spec.jsx (hunk 2 above)
Comment (tlei123) [reply src/applications/simple-forms/21-0845/tests/pages/releaseDuration.unit.spec.jsx:12]: Title fixed.  Thx for catching!

**Comment by tlei123** [comment]
Just made this a Draft PR.  Some content needs fixing first before making this real PR again.

File: src/applications/simple-forms/21-0845/tests/e2e/0845-auth-disclose.cypress.spec.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+      Cypress.config({ waitForAnimations: true, defaultCommandTimeout: 8000 });
+      // Log in if the form requires an authenticated session.
+      // cy.login();
+      cy.intercept('POST', formConfig.submitUrl, { statusCode: 200 });
+      cy.intercept('GET', '/v0/feature_toggles*', featureToggles);
+    },
+
+    // Skip tests in CI until the form is released.
+    // Remove this setting when the form has a content page in production.
+    skip: Cypress.env('CI'),
```
Comment (rhasselle-oddball) [inline src/applications/simple-forms/21-0845/tests/e2e/0845-auth-disclose.cypress.spec.js:60]: let's try to fix this so we aren't skipping it. either as part of this request, or let's create a ticket specifically to fix this

Comment (ndsprinkle) [reply src/applications/simple-forms/21-0845/tests/e2e/0845-auth-disclose.cypress.spec.js:60]: I was going to look into all of the cypress skips after my current ticket, but you're right, we probably should have a ticket to deal with it :D

Comment (tlei123) [reply src/applications/simple-forms/21-0845/tests/e2e/0845-auth-disclose.cypress.spec.js:60]: New ticket opened for this: https://github.com/department-of-veterans-affairs/VA.gov-team-forms/issues/394

**Comment by tlei123** [comment]
New unit-test added for new page, and E2E-test updated for Review-page SoT-alert.  PR now ready for final review & approval.

FYI, new ticket opened to try unskipping E2E-specs in CI: https://github.com/department-of-veterans-affairs/VA.gov-team-forms/issues/394

**Review by rhasselle-oddball — APPROVED** [review]
lgtm. can you update the summary so that it doesn't say do not submit? :)

**Pull request merged**